### ISSUE TITLE
Detect violations for components that conditionally return

### ISSUE THREAD

**Issue by adbutterfield** [kopiro/eslint-plugin-ssr-friendly#25]
I found that violations aren't detected in cases like this:

```jsx
export const SomeComponent = () => {
  const width = window.innerWidth;
  window.addEventListener('resize', () => {
    console.log(document.body.clientWidth);
  });

  document.body.clientWidth;

  return true ? <div /> : null;
};
```
